server/core/config/valid/global_cfg.go

```diff
@@ -147,6 +147,7 @@ type MergedProjectCfg struct {
 	Workspace           string
 	Name                string
 	AutoplanEnabled     bool
+	WhenModified        []string
```

Comment:
We could host the `AutoplanEnabled` and `WhenModified` into it's own struct similar to how we do it for [project](https://github.com/lyft/atlantis/blob/2d3f32d645eac4ce5695750b2c7534598d1d1df6/server/core/config/valid/project.go#L26). However, since `MergedProjConfig` is used everywhere, it'd require a large refactor. I'm fine with making the refactor but wanted to get some thoughts before I go down the path of making changes across numerous files